server/src/com/thoughtworks/go/server/service/PipelineConfigService.java

```diff
@@ -109,6 +111,15 @@ public PipelineConfig getPipelineConfig(String pipelineName) {
         return goConfigService.getConfigForEditing().getPipelineConfigByName(new CaseInsensitiveString(pipelineName));
     }
 
+    public String getPipelineSettingsLink(String pipelineName, Username username) {
```

Comment:
This logic must be in the service, Ideally the view should be checking if the user is admin before rendering the link.

Look at how it's being set in the view
Check usages of https://github.com/gocd/gocd/blob/master/server/src/com/thoughtworks/go/server/web/GoVelocityView.java#L44